qr.go

```diff
@@ -0,0 +1,87 @@
+package txtdirect
+
+import (
+	"encoding/hex"
+	"fmt"
+	"image/color"
+	"net/http"
+	"strconv"
+
+	"github.com/mholt/caddy"
+	qrcode "github.com/skip2/go-qrcode"
+)
+
+// Qr contains Qr code generator's configuration
+type Qr struct {
+	Size            int
+	BackgroundColor string
+	ForegroundColor string
+
+	backgroundColor color.Color
+	foregroundColor color.Color
+}
+
+func (qr *Qr) Redirect(w http.ResponseWriter, r *http.Request) error {
+	Qr, err := qrcode.New(r.Host+r.URL.String(), qrcode.High)
+	if err != nil {
+		return fmt.Errorf("Couldn't generate the Qr instance: %s", err.Error())
+	}
+	if err = qr.ParseColors(); err != nil {
+		return fmt.Errorf("Coudln't parse colors: %s", err.Error())
+	}
+	Qr.BackgroundColor, Qr.ForegroundColor = qr.backgroundColor, qr.foregroundColor
+
+	Qr.Write(qr.Size, w)
+	return nil
+}
+
+func (qr *Qr) ParseColors() error {
+	bg, err := hex.DecodeString(qr.BackgroundColor)
+	if err != nil {
+		return err
+	}
+	qr.backgroundColor = color.RGBA{bg[0], bg[1], bg[2], bg[3]}
+	fg, _ := hex.DecodeString(qr.ForegroundColor)
+	if err != nil {
+		return err
+	}
+	qr.foregroundColor = color.RGBA{fg[0], fg[1], fg[2], fg[3]}
+	return nil
+}
+
+func (qr *Qr) ParseQr(c *caddy.Controller) error {
+	switch c.Val() {
```

Comment:
If we go with being able to configure these things (a lot of configuration options in the long run) we should expose the error recovery.